task.sql

```diff
@@ -1 +1,33 @@
-# Write your SQL code for the database creation here. Good luck! 
+CREATE DATABASE IF NOT EXISTS ShopDB;
+USE ShopDB;
+
+CREATE TABLE IF NOT EXISTS Products (
+    ID INT PRIMARY KEY AUTO_INCREMENT,
+    Name VARCHAR(50) NOT NULL,
+    Description VARCHAR(100) NOT NULL,
+    Price DECIMAL(10, 2) NOT NULL,
+    WarehouseAmount INT NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS Customers (
+    ID INT PRIMARY KEY AUTO_INCREMENT,
+    FirstName VARCHAR(50) NOT NULL,
+    LastName VARCHAR(50) NOT NULL,
+    Email VARCHAR(50) NOT NULL,
+    Address VARCHAR(100) NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS Orders (
+    ID INT PRIMARY KEY AUTO_INCREMENT,
+    CustomerID INT,
+    FOREIGN KEY (CustomerID) REFERENCES Customers(ID) ON DELETE SET NULL,
+    Date DATE NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS OrderItems (
+    ID INT PRIMARY KEY AUTO_INCREMENT,
+    OrderID INT,
```

Comment:
The 'OrderID' column should be nullable since the foreign key constraint uses 'ON DELETE SET NULL'. Consider changing it to 'OrderID INT NULL'.